liquibase-standard/src/main/java/liquibase/analytics/LiquibaseAnalyticsListener.java

```diff
@@ -112,51 +113,43 @@ private synchronized void flush() throws Exception {
+            DumperOptions dumperOptions = new DumperOptions();
+            dumperOptions.setDefaultScalarStyle(DumperOptions.ScalarStyle.DOUBLE_QUOTED);
+            dumperOptions.setWidth(Integer.MAX_VALUE);
+            dumperOptions.setPrettyFlow(true);
+            Yaml yaml = new Yaml(dumperOptions);
+            yaml.setBeanAccess(BeanAccess.FIELD);
 
-                AnalyticsBatch analyticsBatch = AnalyticsBatch.fromLiquibaseEvent(cachedEvents, userId);
-                String jsonInputString = YamlSerializer.removeClassTypeMarksFromSerializedJson(yaml.dumpAs(analyticsBatch, Tag.MAP, DumperOptions.FlowStyle.FLOW));
-                logger.log(logLevel, "Sending anonymous data to Liquibase analytics endpoint. " + System.lineSeparator() + jsonInputString, null);
+            AnalyticsBatch analyticsBatch = AnalyticsBatch.fromLiquibaseEvent(cachedEvents, userId);
+            String jsonInputString = YamlSerializer.removeClassTypeMarksFromSerializedJson(yaml.dumpAs(analyticsBatch, Tag.MAP, DumperOptions.FlowStyle.FLOW));
+            logger.log(logLevel, "Sending anonymous data to Liquibase analytics endpoint. " + System.lineSeparator() + jsonInputString, null);
 
-                IOUtils.write(jsonInputString, conn.getOutputStream(), StandardCharsets.UTF_8);
+            IOUtils.write(jsonInputString, conn.getOutputStream(), StandardCharsets.UTF_8);
 
-                int responseCode = conn.getResponseCode();
-                String responseBody = ExceptionUtil.doSilently(() -> {
-                    return IOUtils.toString(conn.getInputStream());
-                });
-                logger.log(logLevel, "Response from Liquibase analytics endpoint: " + responseCode + " " + responseBody, null);
-                conn.disconnect();
-            } catch (Exception e) {
-                throw new RuntimeException(e);
-            }
-            timedOut.set(false);
+            int responseCode = conn.getResponseCode();
+            String responseBody = ExceptionUtil.doSilently(() -> {
+                return IOUtils.toString(conn.getInputStream(), StandardCharsets.UTF_8);
+            });
+            logger.log(logLevel, "Response from Liquibase analytics endpoint: " + responseCode + " " + responseBody, null);
+            conn.disconnect();
             cachedEvents.clear();
-        });
-        eventThread.start();
-        try {
-            eventThread.join(timeoutMillis);
-        } catch (InterruptedException e) {
-            logger.log(logLevel, "Interrupted while waiting for analytics event processing.", e);
-        }
-        if (timedOut.get()) {
-            logger.log(logLevel, "Timed out while waiting for analytics event processing.", null);
+        } catch (Exception e) {
+            if (e instanceof SocketTimeoutException) {
+                logger.log(logLevel, "Timed out while waiting for analytics event processing.", null);
+            }
+            throw new RuntimeException(e);
```

Comment:
Do we want to re-throw as `RuntimeException`? It will ignore the catch at line 81.

Reply:
Good point, this should probably just be an exception.